### PR TITLE
Fix type hinting for mysql fields in @types/mysql

### PR DESCRIPTION
**Pull request by SystemDisc**
> Also, make array type hinting consistent

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/mysqljs/mysql#performing-queries

### CODE REVIEW

**Comment by SystemDisc** [comment]
@andy-ms, what happens if the author never responds?

File: mysql/index.d.ts [hunk 1]
```diff
@@ -151,24 +151,24 @@ interface IQuery {
...
     on(ev: 'result', callback: (row: any, index: number) => void): IQuery;
     on(ev: 'end', callback: () => void): IQuery;
 }
 
 interface IQueryFunction {
     (sql: string): IQuery;
-    (sql: string, callback: (err: IError, ...args: any[]) => void): IQuery;
-    (sql: string, values: Array<any>): IQuery;
-    (sql: string, values: Array<any>, callback: (err: IError, ...args: any[]) => void): IQuery;
+    (sql: string, callback: (err: IError, results?: any, fields?: IFieldInfo[]) => void): IQuery;
```
Comment (andy-ms) [inline mysql/index.d.ts]: If the argument is always provided, it should not be marked as optional. (If the user doesn't want to use all arguments, they don't have to.)

**Comment by andy-ms** [comment]
If the author hasn't responded in 3 days, ping me and I'll merge.

File: mysql/index.d.ts (hunk 1 above)
Comment (SystemDisc) [inline mysql/index.d.ts]: I'll have to run some tests to see if all arguments are provided for every query type. The mysqljs documentation is a bit lacking.

**Comment by sheetalkamat** [comment]
No response from other authors for over 5 days. Merging.

**Pull request merged**